echopype/mask/api.py

```diff
@@ -17,6 +17,231 @@
+
+    # check the type of fill_value
+    if not isinstance(fill_value, (int, float, np.ndarray, xr.DataArray)):
+        raise TypeError(
+            "The input fill_value must be of type int or " "float or np.ndarray or xr.DataArray!"
+        )
+
+    # make sure that fill_values is the same shape as var_name, if it is an array
+    if isinstance(fill_value, (np.ndarray, xr.DataArray)) and (
+        fill_value.shape != source_ds[var_name].shape
+    ):
+        raise RuntimeError("If fill_value is an array is must be of the same shape as var_name!")
+
+
+def apply_mask(
+    source_ds: Union[xr.Dataset, str, pathlib.Path],
+    var_name: str,
+    mask: Union[
+        Union[xr.DataArray, str, pathlib.Path], List[Union[xr.DataArray, str, pathlib.Path]]
+    ],
+    fill_value: Union[int, float, np.ndarray, xr.DataArray] = np.nan,
+    storage_options_ds: dict = {},
+    storage_options_mask: Union[dict, List[dict]] = {},
+) -> xr.Dataset:
+    """
+    Applies the provided mask(s) to the variable ``var_name``
+    in the provided Dataset ``source_ds``.
+
+    Parameters
+    ----------
+    source_ds: xr.Dataset or str or pathlib.Path
+        Points to a Dataset that contains the variable the mask should be applied to
+    var_name: str
+        The variable name in ``source_ds`` that the mask should be applied to
+    mask: xr.DataArray or str or pathlib.Path or list of xr.DataArray or str or pathlib.Path
+        The mask(s) to apply to the variable specified by ``var_name``. This input can be a
+        single input or list that corresponds to a DataArray or a path to a DataArray. If a path
+        is provided this should point to a zarr or netcdf file with only one array in it.
+    fill_value: int or float or np.ndarray or xr.DataArray, default=np.nan
+        Specifies the value(s) at false indices
+    storage_options_ds: dict, default={}
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``source_ds``
+    storage_options_mask: dict or list of dict, default={}
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``mask``. If ``mask`` is a list, then this input should either
+        be a list of dictionaries or a single dictionary with storage options that
+        correspond to all elements in ``mask`` that are paths.
+
+    Returns
+    -------
+    xr.Dataset
+        A Dataset with the same format of ``source_ds`` with the mask(s) applied to ``var_name``
+
+    Notes
+    -----
+    If the input ``mask`` is a list, then a logical AND will be used to produce the final
+    mask that will be applied to ``var_name``.
+    """
+
+    # validate the source_ds type or path (if it is provided)
+    source_ds, file_type = validate_source_ds_da(source_ds, storage_options_ds)
+
+    if isinstance(source_ds, str):
+        # open up Dataset using source_ds path
+        source_ds = xr.open_dataset(
```

Comment:
While working on another PR I found that `chunks="auto"` will fail for netcdf. This should be replaced with `chunks={}`.